dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -226,4 +264,180 @@ private void checkModelPluralForm(String apiCategory, String model) {
+		return result;
+	}
+	
+	private boolean haveSearchMethods(DSpaceRestRepository repository) {
+		for (Method method : repository.getClass().getMethods()) {
+			SearchRestMethod ann = method.getAnnotation(SearchRestMethod.class);
+			if (ann != null) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/*
+	 * Adapted from org.springframework.data.rest.webmvc.RepositorySearchController.executeQueryMethod(RepositoryInvoker, MultiValueMap<String, Object>, Method, DefaultedPageable, Sort, PersistentEntityResourceAssembler)
+	 */
+	private Object executeQueryMethod(DSpaceRestRepository repository,
+			MultiValueMap<String, Object> parameters, Method method, Pageable pageable, Sort sort,
+			PagedResourcesAssembler assembler) {
+
+		MultiValueMap<String, Object> result = new LinkedMultiValueMap<String, Object>(parameters);
+		MethodParameters methodParameters = new MethodParameters(method, new AnnotationAttribute(Param.class));
+
+		for (Entry<String, List<Object>> entry : parameters.entrySet()) {
+
+			MethodParameter parameter = methodParameters.getParameter(entry.getKey());
+
+			if (parameter == null) {
+				continue;
+			}
+
+			result.put(parameter.getParameterName(), entry.getValue());
+		}
+
+		return invokeQueryMethod(repository, method, result, pageable, sort);
+	}
+	
+	/*
+	 * Adapted from org.springframework.data.repository.support.ReflectionRepositoryInvoker.invokeQueryMethod(Method, MultiValueMap<String, ? extends Object>, Pageable, Sort)
+	 */
+	public Object invokeQueryMethod(DSpaceRestRepository repository, Method method, MultiValueMap<String, ? extends Object> parameters, Pageable pageable,
+			Sort sort) {
+
+		Assert.notNull(method, "Method must not be null!");
+		Assert.notNull(parameters, "Parameters must not be null!");
+
+		ReflectionUtils.makeAccessible(method);
+
+		return ReflectionUtils.invokeMethod(method, repository, prepareParameters(method, parameters, pageable, sort));
+	}
+
+	/*
+	 * Taken from org.springframework.data.repository.support.ReflectionRepositoryInvoker.prepareParameters(Method, MultiValueMap<String, ? extends Object>, Pageable, Sort)
+	 */
+	private Object[] prepareParameters(Method method, MultiValueMap<String, ? extends Object> rawParameters,
+			Pageable pageable, Sort sort) {
+
+		List<MethodParameter> parameters = new MethodParameters(method, PARAM_ANNOTATION).getParameters();
+
+		if (parameters.isEmpty()) {
+			return new Object[0];
+		}
+
+		Object[] result = new Object[parameters.size()];
+		Sort sortToUse = pageable == null ? sort : pageable.getSort();
+
+		for (int i = 0; i < result.length; i++) {
+
+			MethodParameter param = parameters.get(i);
+			Class<?> targetType = param.getParameterType();
+
+			if (Pageable.class.isAssignableFrom(targetType)) {
+				result[i] = pageable;
+			} else if (Sort.class.isAssignableFrom(targetType)) {
+				result[i] = sortToUse;
+			} else {
+
+				String parameterName = param.getParameterName();
+
+				if (!StringUtils.isNotBlank(parameterName)) {
```

Comment:
I think this be replaced by `StringUtils.isBlank`.

Reply:
ok